py/torch_tensorrt/dynamo/lowering/passes/remove_detach.py

```diff
@@ -0,0 +1,25 @@
+import logging
+from typing import Sequence
+
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def remove_detach(
+    gm: torch.fx.GraphModule, sample_inputs: Sequence[torch.Tensor]
+) -> torch.fx.GraphModule:
+    """Remove detach ops in the graph"""
+    count = 0
+    for node in gm.graph.nodes:
+        # node.target = "detach" in torch.compile workflow
+        if node.target == torch.ops.aten.detach.default or node.target == "detach":
+            # Detach node has only one input
+            node_input = node.all_input_nodes[0]
+            node.replace_all_uses_with(node_input)
+            gm.graph.erase_node(node)
+            count += 1
+
+    logger.debug(f"Removed {count} detach nodes:\n{gm.graph}")
```

Comment:
Consider replacing with the following, to clean up graph + avoid logging if no changes were made to the graph
```python
    if count > 0:
        gm = clean_up_graph_after_modifications(gm)
        logger.debug(f"Removed {count} detach nodes:\n{gm.graph}")

    return gm
```

Reply:
I've intentionally not used it since `clean_up_graph_after_modifications` will invoke `gm.recompile()` which I'm avoiding. The next step of this pass is getting ep.module() which calls recompile inside it.